nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/manifest/StandardRuntimeManifestService.java

```diff
@@ -228,23 +227,18 @@ private static ControllerServiceDefinition getManifestControllerServiceDefinitio
         return definition;
     }
 
-    private static List<AllowableValue> getManifestAllowableValues(final PropertyDescriptor propertyDescriptor) {
-        if (propertyDescriptor.getAllowableValues() == null) {
-            return List.of();
-        }
-
-        final List<AllowableValue> allowableValues = new ArrayList<>();
-        for (final org.apache.nifi.components.AllowableValue allowableValue : propertyDescriptor.getAllowableValues()) {
-            final AllowableValue manifestAllowableValue = new AllowableValue();
-            manifestAllowableValue.setDescription(allowableValue.getDescription());
-            manifestAllowableValue.setValue(allowableValue.getValue());
-            manifestAllowableValue.setDisplayName(allowableValue.getDisplayName());
-            allowableValues.add(manifestAllowableValue);
-        }
-        return allowableValues;
+    private static List<AllowableValue> getAllowableValues(final PropertyDescription propertyDescription) {
+        return Optional.ofNullable(propertyDescription.getAllowableValues()).orElse(List.of())
```

Comment:
This should use `Collections.emptyList()` to avoid unnecessary creation:
```suggestion
        return Optional.ofNullable(propertyDescription.getAllowableValues()).orElse(Collections.emptyList())
```

Reply:
List.of() does not create a new instance, it returns a static constant immutable list just like Collections.emptyList(). I prefer the former due it's shortness and conciseness.

Reply:
Thanks for noting that important detail, I agree, leaving it as `List.of()` sounds good.